### PR TITLE
Sync Skipper client changes

### LINKED ISSUE #1947: Update Skipper client delete request
We need to update the changes from https://github.com/spring-cloud/spring-cloud-skipper/pull/475

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Update release delete request changes
  - Use boolean path variable instead of DeleteProperties
  - Update tests
 - Remove release history --max option
  - Remove the references of max revisions from history command, client and controller side
 - Fix Release.setManifest changes

Resolves #1947

### CODE REVIEW

**Comment by ilayaperumalg** [comment]
Rebased.

File: spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/StreamDeploymentController.java [hunk 1]
```diff
@@ -171,13 +170,6 @@ public String manifest(@PathVariable("name") String name, @PathVariable("version
 		return this.streamService.manifest(name, version);
 	}
 
-	@RequestMapping(path = "/history/{name}/{max}", method = RequestMethod.GET)
```
Comment (tzolov) [inline spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/StreamDeploymentController.java]: why is the history entry point removed?

Comment (ilayaperumalg) [reply spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/StreamDeploymentController.java]: sorry, this was my mistake. In Skipper's ReleaseController we got rid of the entire REST endpoint as it is served from the repository. But, here we use an explicit REST endpoint and it is required. will re-add.

**Comment by tzolov** [comment]
squashed and merged at 6e2f0aaa

**Pull request closed without merging**